src/gengowatcher/watcher.py

```
@@ -2442,19 +2453,38 @@ def run(self):
+            gateway_url = str(
+                self.config.get(
+                    "WebSocket", "gateway_url", fallback="http://127.0.0.1:8000"
+                )
+                or "http://127.0.0.1:8000"
             )
-            self.websocket_status = "Gateway Connected"
-            self.logger.info(f"WebSocket using external gateway at {gateway_url}")
-        elif self.config.get("WebSocket", "enable_websocket"):
+            health_url = f"{gateway_url.rstrip('/')}/health"
+            try:
+                with urlopen(health_url, timeout=1.0) as response:
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)